flair/datasets/sequence_labeling.py

```diff
@@ -443,6 +434,167 @@ def __init__(
             **corpusargs,
         )
 
+    @staticmethod
+    def get_token_level_label_of_each_token(sentence: Sentence, label_type: str) -> List[str]:
+        """Generates a label for each token in the sentence. This function requires that the labels corresponding to the label_type are token-level tokens.
+
+        Args:
+            sentence: a flair sentence to generate labels for
+            label_type: a string representing the type of the labels, e.g., "pos"
+        """
+        list_of_labels = ["O" for _ in range(len(sentence.tokens))]
+        for label in sentence.get_labels(label_type):
+            label_token_index = label.data_point._internal_index
+            list_of_labels[label_token_index - 1] = label.value
+        return list_of_labels
+
+    @staticmethod
+    def get_span_level_label_of_each_token(sentence: Sentence, label_type: str) -> List[str]:
+        """Generates a label for each token in the sentence in BIO format. This function requires that the labels corresponding to the label_type are span-level tokens.
+
+        Args:
+            sentence: a flair sentence to generate labels for
+            label_type: a string representing the type of the labels, e.g., "ner"
+        """
+        list_of_labels = ["O" for _ in range(len(sentence.tokens))]
+        for label in sentence.get_labels(label_type):
+            tokens = label.data_point.tokens
+            start_token_index = tokens[0]._internal_index
+            list_of_labels[start_token_index - 1] = f"B-{label.value}"
+            for token in tokens[1:]:
+                token_index = token._internal_index
+                list_of_labels[token_index - 1] = f"I-{label.value}"
+        return list_of_labels
+
+    @staticmethod
+    def write_dataset_to_file(
```

Comment:
This could also be a non-static method of ColumnDataset, directly callable in case someone wants to serialize one dataset only.